### ISSUE TITLE
[vsc-extension] Stencil VSCode extension

### ISSUE THREAD

**Issue by techsavvyash** [SamagraX-Stencil/stencil#7]
## Description

The aim is to create a first version of a VSCode extension for `stencil` in order to provide a better DX. 
The first feature we are focusing on right now is:
- [ ] to verify the data to be seeded in the DB present in a CSV/JSON file against the model generated by the `prisma-client`.

**Comment by Apurv428** [comment]
Hello @RyanWalker277, I'm keen on taking up this issue. I would like more information, particularly regarding step 2, which involves implementing the data verification feature. This feature aims to enable users to validate data from CSV/JSON files against the Prisma client-generated model. Can you provide more guidance to help me proceed with this task?

**Comment by RyanWalker277** [comment]
Hey @Apurv428, so basically the VS-Code extension should be able to seed the data in the db. The user will provide a CSV including the data to be seeded. It should then verify if the CSV is according to our prisma model. If the validation succeeds, then it should seed the csv data into the database in the appropriate table.

**Comment by Apurv428** [comment]
Thanks for the explanation. So, according to my understanding
The user uploads a CSV, we validate its compatibility with our Prisma model, and if it aligns correctly, we insert it into the necessary database table.

Please correct me if I am wrong.

Do you have any recommendations on how to begin with this? Also, could you share the database format to get acquainted with it?

**Comment by RyanWalker277** [comment]
Yes the understanding is correct. You can refer to this to begin:

https://www.freecodecamp.org/news/making-vscode-extension/

The database format is provided along with the code in the [`schema.prisma`](https://github.com/SamagraX-Stencil/stencil/blob/5cb79173c6f8585b952ff462b4831bc877b3d7a2/sample/01-all-features/prisma/schema.prisma#L3)

**Comment by Apurv428** [comment]
Thank you for providing the reference. If I have any questions, I'll reach out to you.

**Comment by Apurv428** [comment]
Hi @RyanWalker277 
Could you please provide more details on the first point of the acceptance criteria, specifically regarding the extension setup? Are you referring to a basic setup similar to configuring a VSCode extension, or does it involve additional steps or considerations?

**Comment by RyanWalker277** [comment]
Hey @Apurv428, in the first iteration it should be a basic extension which provides the functionality to seed db from csv data. There would be more features and enhancements after this feature is initialised. Let me know when you are ready to create a pull request, I'll create a separate repository for the extension. You'll have to create a PR on that.

**Comment by Apurv428** [comment]
Okay, I have created a basic extension and am working on the upload functionality.
Thank you!

**Comment by Apurv428** [comment]
Hi @RyanWalker277 ,  how can I get the format of database? Is it same as the Role model?

**Comment by techsavvyash** [comment]
Hey @Apurv428 ,
The database schema would be dynamic, we'll need to parse the schema.prisma file to figure that out. The Role schema is just for reference.

**Comment by Apurv428** [comment]
So, similar to Role model, there may be other models mentioned in that file?

**Comment by techsavvyash** [comment]
Yes

**Comment by Apurv428** [comment]
Sure!

**Comment by Apurv428** [comment]
Hi,
Could y'll provide the following details about database seeding?

Where should we retrieve the database details from? Are they obtained in the same way from Prisma? Could you share the env variable values related to the database? Additionally, will the table names correspond to the model names?

**Comment by techsavvyash** [comment]
Hey @Apurv428,
Database seeding is still something we are figuring out how to implement into stencil, we can add that as a part of this extension once it has been discussed and a seeding strategy has been agreed upon.

**Comment by Apurv428** [comment]
Okay, thank you!

**Comment by Yash-Sajwan24** [comment]
@techsavvyash  I understand the problem, and I believe I have the necessary skills. Could I have the opportunity to work on this?

**Comment by Abhi-0001** [comment]
is it still available to work on ? Can I start work on it?
If yes, then what is the current status of extension?

**Comment by techsavvyash** [comment]
@Abhi-0001 the first version of the vs code extension was very generously contributed by @Apurv428. The issue is still open to work on by adding more functionality enhancing the developer experience.

**Comment by Abhi-0001** [comment]
Thanks @techsavvyash, so Can you give me instructions about setting the code up in my machine (as this is my first time in vscode extension development, though I know typescript) and the requirements I would have to full fill for next version.

**Comment by techsavvyash** [comment]
Refer #78

**Comment by thekavikumar** [comment]
@techsavvyash by adding some more features, will I be getting the points?

**Comment by techsavvyash** [comment]
yes @thekavikumar

**Comment by thekavikumar** [comment]
provide immediate feedback on whether the data matches the model, reducing errors and debugging time.

@techsavvyash sir is this feature good to implement?

**Comment by VedantKhairnar** [comment]
@techsavvyash 
Is the PR and the work around this issue completed? If yes, will you pls close the issue? Thanks.

**Comment by techsavvyash** [comment]
@VedantKhairnar this is a long running ticket.

**Comment by devXpraAddy** [comment]
@techsavvyash I would like to resolve this issue under C4GT community, please assign this issue to me
